tuta-sdk/rust/src/crypto/entity_facade.rs

```diff
@@ -0,0 +1,304 @@
+                    self.resolve_default_value(model_value.value_type.to_owned())
+                }
+            };
+
+            return Ok((element_value, final_ivs, errors));
+        }
+
+        return Ok((value.clone(), final_ivs, errors));
+    }
+
+    fn extract_iv(&self, encrypted_data: &[u8]) -> [u8; IV_BYTE_SIZE] {
+        let mut iv_bytes: [u8; IV_BYTE_SIZE] = Default::default();
+        iv_bytes.clone_from_slice(&encrypted_data[0..IV_BYTE_SIZE]);
+        iv_bytes
+    }
+
+    fn resolve_default_value(&self, value_type: ValueType) -> ElementValue {
+        return match value_type {
+            ValueType::String => ElementValue::String(String::new()),
+            ValueType::Number => ElementValue::Number(0),
+            ValueType::Bytes => ElementValue::Bytes(Vec::new()),
+            ValueType::Date => ElementValue::Date(SystemTime::UNIX_EPOCH),
+            ValueType::Boolean => Bool(false),
+            ValueType::CompressedString => ElementValue::String(String::new()),
+            _ => panic!("Invalid type")
+        };
+    }
+
+    fn parse_decrypted_value(&self, value_type: ValueType, bytes: &[u8]) -> Result<ElementValue, ApiCallError> {
+        return match value_type {
+            ValueType::String => Ok(ElementValue::String(String::from_utf8_lossy(bytes).to_string())),
+            ValueType::Number => {
+                if bytes.len().eq(&0) {
+                    return Ok(ElementValue::Null);
+                } else {
+                    let bytes = match bytes.try_into() {
+                        Ok(bytes) => bytes,
+                        Err(_) => return Err(ApiCallError::InternalSdkError { error_message: "Failed to parse bytes slice".to_string() })
+                    };
+                    Ok(ElementValue::Number(i64::from_be_bytes(bytes)))
+                }
+            }
+            ValueType::Bytes => Ok(ElementValue::Bytes(bytes.to_vec())),
+            ValueType::Date => {
+                let bytes = match bytes.try_into() {
+                    Ok(bytes) => bytes,
+                    Err(_) => return Err(ApiCallError::InternalSdkError { error_message: "Failed to parse bytes slice".to_string() })
+                };
+                Ok(ElementValue::Date(SystemTime::UNIX_EPOCH + Duration::from_millis(u64::from_be_bytes(bytes))))
+            }
+            ValueType::Boolean => {
+                let value = if bytes.eq(&[0x01]) {
+                    true
+                } else if bytes.eq(&[0x00]) {
+                    false
+                } else {
+                    return Err(ApiCallError::InternalSdkError { error_message: "Failed to parse boolean bytes".to_string() });
+                };
+
+                Ok(Bool(value))
+            }
+            ValueType::CompressedString => Ok(ElementValue::String(String::from_utf8_lossy(bytes).to_string())),
+            _ => panic!("Failed to parse bytes into ElementValue")
```

Comment:
we do not support compressed strings yet, idk if we should just error out on them

Reply:
For now I replaced with an error. I'm wondering if we should panic with unimplemented